### DISCUSSION TITLE
Unable to connect to a branch using access token

### DISCUSSION THREAD

**Discussion by edgrosvenor** [planetscale/discussion#293] [Bug Reports]
In my PHP app with the access token environment variables set up, I am able to create a new branch in my app like so:

`pscale branch create $database $branch --org $org`

This indicates to me that the environment variables are set up correctly and that my general usage of the cli is correct. However, when I attempt to connect to the newly created branch using this

`pscale connect $database $branch --org $org`

I get an error that `Error: database $database and branch $branch does not exist in organization $org`

When I run the same command in a separate terminal that is authenticated using `pscale auth login` I am greeted with a successful connection.

I believe the token has the required permissions.

<img width="645" alt="CleanShot 2022-09-27 at 15 36 38@2x" src="https://user-images.githubusercontent.com/1053395/192541420-e18cd51b-3840-4a18-8e02-fe37c0b94098.png">

`read_deploy_request, read_comment, read_branch, delete_branch_password, delete_branch, create_deploy_request, create_comment, create_branch, connect_branch, approve_deploy_request`

**Comment by orware** [comment]
Hi @edgrosvenor!

Thanks for reaching out about this situation!

Would you be able to take this over to https://support.planetscale.com/ when you have a moment since there are likely a few extra details about the PHP code itself I might want to take a closer look at and try running some tests on with the `pscale` CLI to confirm if this is a bug or not and it'll be easier to do that back and forth via a ticket and we can circle back here with the resolution. 

Thank you!

**Comment by edgrosvenor** [comment]
@orware I will do that. I'm in Switzerland so my day is done, but I'll head over there in the morning and get started. Thanks!

**Reply by orware** [reply]
No problem @edgrosvenor!

I have completed a few tests on my end as well and the only time I received something similar to what you described with this error message:

```
Error: database $database and branch $branch does not exist in organization $org
```

Occurred when I attempted to use a database/branch that the service token was not allowed to use yet within the organization. Once I switched back to a database/branch the service token was allowed to use the error went away.

I am also assuming that the error message above didn't literally display `$database` / `$branch` / `$org` in the error either since that would be an indication of the placeholder values not getting replaced correctly, but I'm assuming you left the placeholders in your example just to keep things generic here in your initial report above.

I initially thought the error could be related to the `connect_production_branch` permission not being enabled but that leads to a different error message if you try to connect to a production branch:

```
Error: User does not have permission to perform this action.
```

On my end to run the tests I ended up first using `pscale auth logout` and then I used examples similar to yours above but directly included the service token and ID values in some quick test calls:

```
pscale branch create <DATABASE_NAME> <BRANCH_NAME> --org <ORGANIZATION_NAME> --service-token <SERVICE_TOKEN> --service-token-id <SERVICE_TOKEN_ID>
```

```
pscale connect <DATABASE_NAME> <BRANCH_NAME> --org <ORGANIZATION_NAME> --service-token <SERVICE_TOKEN> --service-token-id <SERVICE_TOKEN_ID>
```

So you may be able to do something similar as you continue the testing process.

If the above isn't helpful in getting to a resolution, creating the ticket may be the next best thing so we can continue trying to figure out the issue!

**Comment by edgrosvenor** [comment]
@orware This is fun. I changed absolutely nothing in my code and it works this morning. The command times out in my PHP script for whatever reason, but I can see in the audit log that the connection was successful. What I was hoping to do in the PHP script is request json with the `--format json` flag, but even if I do that in my logged in terminal session that times out.

Either way, my initial bug report seems to be invalid, so I'll see if I can get past my current issue and if not I'll open a ticket.
